tests/unit/test_notifications.py

```diff
@@ -103,6 +118,27 @@ def test_file_upload_publish_raises(self, exception):
             block.fire_file_upload_notification(self.notifications_service_mock)
             patched_exception_logger.assert_called_once_with(exception)
 
+
+@ddt.ddt
+class TestActivityNotificationsMixin(TestCase, TestWithPatchesMixin):
+    workgroup1 = make_workgroup([1, 2, 3])
+
+    @staticmethod
+    def _raise(exception):
+        raise exception
+
+    def setUp(self):
+        self.notifications_service_mock = mock.Mock()
+        self.notifications_service_mock.get_notification_type = mock.Mock(side_effect=get_notification_type)
+        self.notifications_service_mock.bulk_publish_notification_to_users = mock.Mock()
+        self.notifications_service_mock.bulk_publish_notification_to_scope = mock.Mock()
+
+    def _get_call_args(self, target):
+        self.assertTrue(target.called)
+        self.assertEqual(len(target.call_args_list), 1)
+        args, _kwargs = target.call_args
+        return args
```

Comment:
@ziafazal it would be great to extract these helper methods to common base class, instead of duplicating them.